### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by suzi216**
## **OOP school library: preserve data**

- Save all the data (books, people, and rentals) into JSON files.
- On startup of your app load all data from files
- No linter errors

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -31,6 +32,7 @@ def list(items, choose)
   end
 
   def list_all_books
+    print @rentals
```
Comment (julie-ify) [inline app.rb]: - [ ] Kindly remove this printing of `@rentals` here.

File: app.rb [hunk 2]
```diff
@@ -2,6 +2,7 @@
 require './student'
 require './teacher'
 require './rental'
+require 'json'
```
Comment (julie-ify) [inline app.rb:5]: - [ ] When I delete the JSON files and try to start the app it crashes with the error below. Kindly ensure you handle the scenario when the JSON files do not exist - like creating the JSON files and initializing an empty array in them if they do not exist.

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Suzi216 and @Htetaungkyaw71,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
## STATUS APPROVED ✔️ 🎉 👍 

Hi @Suzi216 & @Htetaungkyaw71, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [ ] Descriptive PR ✔️ 
- [ ] No linters errors ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**